### DISCUSSION TITLE
More examples

### DISCUSSION THREAD

**Discussion by peepo5** [fabianboesiger/ftx#64] [Ideas]
It would be nice to have a few more examples of use for this module.
While developing new additions, it will be much easier if I have some references to examples.

For example, I want to add trigger orders but It is giving me errors presumably because I do not know how to format the inputs while using the function.

Some examples I would like:
- Order example
- List market names (in a mod, not main)
- Error handling with other error types (using anyhow crate would work best)
- Examples of math to info with rust_decimal crate for handling numbers

**Comment by fabianboesiger** [comment]
In general, the inputs of the functions, for example the `place_order` function, are the same as specified in the [FTX API docs](https://docs.ftx.com/#place-order).

I'm not sure what you mean by list market names "in a mod, not main", but I just added a new example that lists all market names.

Error handling is not really something specific to this project, if you want to learn more about error handling in Rust, you can do so by reading guides on the internet.

For handling numbers using rust_decimal, you can refer to the [docs](https://docs.rs/rust_decimal/1.16.0/rust_decimal/).

If you have more specific questions, you can maybe post your code that causes errors and I'll try to help.

**Reply by peepo5** [reply]
what i mean with the "in a mod, not main" would just be by declaring a module then using a function inside of it to return, lets say, prices of BTC. this would help with many cases where people want to do such a thing.

I would also advice to give an example with error handling (which I can provide) so it does not panic when ftx gives a bad return.

Where you say, in general the inputs are the same as the docs, I tried providing "buy" as a side option when doing an order and it would not let me. I think a full example of an order input would be of substantial help.

**Comment by fabianboesiger** [comment]
To provide the side for placing an order, you have to use the type `Side::Buy` or `Side::Sell`. You can look up what the expected fields of a method are in the docs: https://docs.rs/ftx/0.3.1/ftx/rest/struct.Rest.html#method.place_order

In any case, I will try to add more examples in the future.

Edit: I just realized that the docs aren't up to date. You can view the latest docs for ftx by running `cargo doc --open` in the cloned ftx repo.

**Reply by peepo5** [reply]
:+1: I will add examples as I use the module more too.

**Comment by peepo5** [comment]
Examples on my project:
[termcrypt](https://github.com/termcrypt/termcrypt)
@fabianboesiger
